hydromt/models/model_api.py

```diff
@@ -28,6 +30,14 @@
 
 logger = logging.getLogger(__name__)
 
+DeferedFileClose = TypedDict(
+    "DeferedFileClose",
+    {"ds": xr.Dataset, "org_fn": str, "tmp_fn": str, "close_attempts": int},
+)
+
+
+TMP_DATA_DIR = TemporaryDirectory()
```

Comment:
This creates a tempdir every time we import hydromt right? Is that what we want?
Could we also create a temp dir only when necessary in the _write_nc method. Possibly even located inside the model root?